compel/plugins/std/log.c

```diff
@@ -54,10 +60,19 @@ static void sbuf_log_init(struct simple_buf *b)
 
 	if (start.tv_sec != 0) {
 		struct timeval now;
+		/* temporary variable to store now */
+		struct timeval temp;
 
 		std_gettimeofday(&now, NULL);
+
+		if(relative_timestamps)
+			temp = now;
+
 		timediff(&start, &now);
 
+		if(relative_timestamps)
+			start = temp;
```

Comment:
Is this function called from a few threads concurrently? If I remember correctly, this can happen when threads are restored. It means that we need to use locks or atomic operations here. Or in an ideal case, we can think about how to make per-thread relative timestamps.

Reply:
My implementation of `relative_timestamp` option is similar to `cur_loglevel`. But there were no locks used for `cur_loglevel`.
So, I don't think `relative_timestamps` is being called concurrently here.

Reply:
cur_loglevel is never changed, so we can read it concurrently without locks. This case is different...

Look at the code where we fork threads:
https://github.com/checkpoint-restore/criu/blob/criu-dev/criu/pie/restorer.c#L1811

Reply:
Ok, I will try to make the changes.